build-wheelhouse/action.yml

```diff
@@ -153,16 +161,32 @@ runs:
       shell: bash
       env:
         TARGET: ${{ inputs.target }}
+        WORKING_DIRECTORY: ${{ inputs.working-directory }}
       run: |
-        echo "install_target=$( [[ "${TARGET}" == '' ]] && echo '.' || echo ".[${TARGET}]")" >> ${GITHUB_OUTPUT}
+        echo "install_target=$( [[ "${TARGET}" == '' ]] && echo "${WORKING_DIRECTORY}" || echo "${WORKING_DIRECTORY}[${TARGET}]")" >> ${GITHUB_OUTPUT}
         echo "wheelhouse_target=$( [[ "${TARGET}" == '' ]] && echo 'wheelhouse' || echo "${TARGET}-wheelhouse")" >> ${GITHUB_OUTPUT}
 
-    - name: "Install the library"
+    - name: "Identify the build system"
+      id: build-system
       shell: bash
+      run: |
+        if [[ -f "pyproject.toml" ]] && grep -q 'build-backend = "poetry\.core\.masonry\.api"' "pyproject.toml"; then
+          echo "BUILD_BACKEND=$(echo 'poetry')" >> ${GITHUB_OUTPUT}
+        else
+          echo "BUILD_BACKEND=$(echo 'pip')" >> ${GITHUB_OUTPUT}
+        fi
+
+    - name: "Install the library"
       env:
         INSTALL_TARGET: ${{ steps.specific-target-requested.outputs.install_target }}
+        BUILD_BACKEND: ${{ steps.build-system.outputs.BUILD_BACKEND }}
+      shell: bash
       run: |
-        python -m pip install "${INSTALL_TARGET}"
+        if [[ "${BUILD_BACKEND}" == 'poetry' ]]; then
+          poetry install ${INSTALL_TARGET}
```

Comment:
We should check that we are doing this as well in other actions -- I'm thinking mostly the testing and docs actions, although I think it's already the case

Reply:
It looks like the other actions are using poetry install commands. I moved the changes for the working directory in https://github.com/ansys/actions/pull/807.